src/util/lie/so3.cpp

```diff
@@ -0,0 +1,18 @@
+#include "se3.hpp"
+#include <stdexcept>
+
+SO3::SO3(Eigen::Quaterniond const& quaternion) : mQuaternion(quaternion) {
+    if (std::fabs(mQuaternion.norm() - 1.0) > std::numeric_limits<double>::epsilon()) {
```

Comment:
this is going to become a problem for someone eventually because SO3 isn't renormalizing quaterrnions right now. Throwing an exception is good since its loud at least, maybe also drop a TODO or note in there saying that if you end up here you probably need to implement normalization.

Reply:
https://stackoverflow.com/questions/11667783/quaternion-and-normalization

Reply:
Good point. I wonder how `Eigen::Transform` deals with this